accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,87 @@
+# Simplify .NET Output Paths
+
+We'd like to simplify the output paths for .NET 8.
+
+Currently, the default output path for a .NET project includes folders for the `TargetFramework` and the `RuntimeIdentifier` (if it is set).  Additionally, the publish output goes in a `publish` subfolder of the output folder.  The resulting paths are as follows:
+
+- `bin\<Configuration>\<TargetFramework>` - Build output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>` - Build output with `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\publish` - Publish output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>\publish` - Publish output with `RuntimeIdentifier`
+
+This is rather complicated and inconsistent, which can make for a poor first impression of the .NET platform.
+
+We'd like to try to improve the output path structure.  Desired qualities include:
+
+- Simple and consistent
+- Avoid cases where one output path is nested inside of another one
+- Avoid excessively deep directory hierarchies
+- Get rid of `obj` folder in project root
+
+## Proposed behavior
+
+The new output path format will consist of the following 3 nested folders:
+
+- `artifacts` - All output will go under this folder
```

Comment:
> for what it's worth, [the Visual Studio default .gitignore already ignores any artifacts directory](https://github.com/github/gitignore/blob/e5323759e387ba347a9d50f8b0ddd16502eb71d4/VisualStudio.gitignore#L63), and has done so [since 2015](https://github.com/github/gitignore/commit/467e257fdb71b6b920dde08ef16337e80ef91bb6). Though not everyone uses git, and not all git users use the default .gitignore.

from @zivkan above https://github.com/dotnet/designs/pull/281#discussion_r1028610642

Reply:
`artifacts` has conveniently been [included](https://github.com/dotnet/designs/pull/281#discussion_r1028610642) in the default Visual Studio .gitignore file since 2015.  That doesn't cover all .NET projects, but it mitigates the impact to .gitignore a lot.

Reply:
just curious, how widely is `artifacts` used today eg in other stacks / in OSS ? wondering how it ended up in that gitignore. Ideally we'd use a name already widely used (ideally not just Microsoft recognized)

Reply:
Is this really the main reason to choose `artifacts` as the name? It seems other, shorter names are options even if they may not be in `.gitignore` today.

Something like:

- `out/bin/<pivot>`
- `out/temp/<pivot>`

Seems just as clear, but much shorter. If lining up with `.gitignore` is an intended goal, then it should be called out as such. I think it would eliminate some lines of questioning regarding the names proposed.